### PR TITLE
CHI-2539: Frontend - transfer draft cases (NZ)

### PR DESCRIPTION
**Pull request by acedeywin**
Primary reviewer:

## Description

- New feature: This ticket handles the follow:

- Remove code that disconnects case from contact during transfer
- If counsellor A has connected a case to contact and then transfers to counsellor B, then A may not have permission to view/disconnect/edit that case in the future
- Handle edge case if someone can view a contact but NOT the case associated with it

### Checklist
- [x] Corresponding issue has been opened
- [ ] New tests added
- [ ] Feature flags added
- [ ] Strings are localized
- [ ] Tested for chat contacts
- [ ] Tested for call contacts

### Related Issues
Fixes [CHI-2539](https://tech-matters.atlassian.net/browse/CHI-2539)

### Verification steps

[CHI-2539]: https://tech-matters.atlassian.net/browse/CHI-2539?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: plugin-hrm-form/src/components/caseMergingBanners/ContactAddedToCaseBanner.tsx [hunk 1]
```diff
@@ -85,13 +99,20 @@ const ContactAddedToCaseBanner: React.FC<Props> = ({
...
+        permission={!canEditContact && 'none'}
+        onClick={() => canEditContact && viewCaseDetails(connectedCase)}
+      >
         <Template code="Case-CaseNumber" />
         {caseId}
       </CaseLink>
-      <BannerActionLink type="button" onClick={handleRemoveContactFromCase}>
-        <Template code="CaseMerging-RemoveFromCase" />
-      </BannerActionLink>
+      {canEditContact && (
```
Comment (stephenhand) [inline plugin-hrm-form/src/components/caseMergingBanners/ContactAddedToCaseBanner.tsx:111]: PermissionActions.EDIT_CONTACT is not the correct action to check here, it should be PermissionActions.REMOVE_CONTACT_FROM_CASE and the can function needs to be passed the case object

File: plugin-hrm-form/src/components/caseMergingBanners/ContactRemovedFromCaseBanner.tsx [hunk 2]
```diff
@@ -65,30 +66,38 @@ const ContactRemovedFromCaseBanner: React.FC<Props> = ({
...
-      <Close />
-    </HeaderCloseButton>
-  </BannerContainer>
-);
+}) => {
+  const can = React.useMemo(() => {
+    return getInitializedCan();
+  }, []);
+
+  const canUndo = showUndoButton && savedContact?.id && can(PermissionActions.EDIT_CONTACT, savedContact);
```
Comment (stephenhand) [inline plugin-hrm-form/src/components/caseMergingBanners/ContactRemovedFromCaseBanner.tsx:74]: This needs to be PermissionActions.ADD_CONTACT_TO_CASE

File: plugin-hrm-form/src/components/contact/ContactDetailsHome.tsx [hunk 3]
```diff
@@ -145,6 +145,8 @@ const ContactDetailsHome: React.FC<Props> = function ({
     [],
   );
 
+  console.log('savedContact is contact home', savedContact);
```
Comment (stephenhand) [inline plugin-hrm-form/src/components/contact/ContactDetailsHome.tsx:148]: Remove before merging ;-)

**Review by stephenhand — COMMENTED** [review]
The permission actions you are checking need changing, otherwise looks good

File: plugin-hrm-form/src/components/caseMergingBanners/ContactAddedToCaseBanner.tsx [hunk 4]
```diff
@@ -85,13 +101,21 @@ const ContactAddedToCaseBanner: React.FC<Props> = ({
       <Text>
         <Template code="CaseMerging-ContactAddedTo" />
       </Text>
-      <CaseLink type="button" onClick={() => viewCaseDetails(connectedCase)} data-fs-id="LinkedCase-Button">
+      <CaseLink
```
Comment (stephenhand) [inline plugin-hrm-form/src/components/caseMergingBanners/ContactAddedToCaseBanner.tsx:104]: Same fix from other PR needs applying here

Comment (acedeywin) [reply plugin-hrm-form/src/components/caseMergingBanners/ContactAddedToCaseBanner.tsx:105]: @stephenhand - It's done

File: plugin-hrm-form/src/components/caseMergingBanners/ContactAddedToCaseBanner.tsx [hunk 5]
```diff
@@ -85,13 +102,21 @@ const ContactAddedToCaseBanner: React.FC<Props> = ({
...
+        type="button"
+        color={!canEditContact && '#000'}
+        permission={!canEditContact && 'none'}
+        onClick={() => canEditContact && canViewCase && viewCaseDetails(connectedCase)}
```
Comment (stephenhand) [inline plugin-hrm-form/src/components/caseMergingBanners/ContactAddedToCaseBanner.tsx:109]: This still doesn't look right? I thought we just need to check `canViewCase` in color, permission and onClick?

File: plugin-hrm-form/src/components/caseMergingBanners/ContactAddedToCaseBanner.tsx (hunk 4 above)
Comment (stephenhand) [reply plugin-hrm-form/src/components/caseMergingBanners/ContactAddedToCaseBanner.tsx:105]: This check is still not correct. This link should be active if the user has view permissions, they don't need remove permissions as well.

Comment (acedeywin) [reply plugin-hrm-form/src/components/caseMergingBanners/ContactAddedToCaseBanner.tsx:105]: @stephenhand  - Fixed it.

**Review by stephenhand — APPROVED** [review]
